mne_bids/write.py

```diff
@@ -1125,6 +1126,7 @@ def write_raw_bids(raw, bids_basename, bids_root, events_data=None,
     if raw.preload is not False:
         raise ValueError('The data should not be preloaded.')
 
+    bids_root = _ensure_pathlike(bids_root)
```

Comment:
this should also be in the reader no?

Reply:
Actually no. It's only in `write_bids_raw` bc this function returns the `bids_root`. Otherwise it would be sufficient to only call `_ensure_pathlike` in `utils.make_bids_folders`.

The readers pass the paths directly to `mne.io`, and MNE does support Path-like objects already. (And if we come across a situation where it doesn't, that's something that would need fixing).